controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -230,7 +230,7 @@ func (r *RegisteredClusterReconciler) createManagedCluster(regCluster *singapore
 
 	// check if managedcluster is already exists
 	managedClusterList := &clusterapiv1.ManagedClusterList{}
-	if err := r.HubClusters[0].Client.List(context.Background(), managedClusterList, client.MatchingLabels{RegisteredClusterNamelabel: regCluster.Name, RegisteredClusterNamespacelabel: regCluster.Namespace}); err != nil {
+	if err := r.HubClusters[0].APIReader.List(context.Background(), managedClusterList, client.MatchingLabels{RegisteredClusterNamelabel: regCluster.Name, RegisteredClusterNamespacelabel: regCluster.Namespace}); err != nil {
```

Comment:
same why apireader? I think we don't  need apireader here. MCE support small no of managedclusters, we should read it through cache.